lib/ddtrace/contrib/redis/patcher.rb

```diff
@@ -93,6 +79,18 @@ def call_pipeline(*args, &block)
 
               response
             end
+
+            def datadog_pin
```

Comment:
This fixes a bug where a Redis client initialized before being patched wouldn't receive trace configuration, and thus wouldn't send traces. By lazy initializing the `Pin`, we can ensure it's available at trace time.